src/main/java/htsjdk/tribble/gff/Gff3Codec.java

```diff
@@ -229,11 +225,29 @@ static private Gff3BaseData parseLine(final String line, final int currentLine)
         }
     }
 
-    public Collection<SequenceRegion> getSequenceRegions() {
-        return sequenceRegionMap.values();
+    /**
+     * Get list of sequence regions parsed by the codec.
+     * @return list of sequence regions
+     */
+    public List<SequenceRegion> getSequenceRegions() {
+        return Collections.unmodifiableList(new ArrayList<>(sequenceRegionMap.values()));
     }
 
-    public List<String> getComments() {return comments;}
+    /**
+     * Gets map from line number to comment found on that line.  The text of the comment EXCLUDES the leading # which indicates a comment line.
+     * @return Map from line number to comment found on line
+     */
+    public Map<Integer, String> getCommentsWithLineNumbers() {
+        return Collections.unmodifiableMap(new LinkedHashMap<>(commentsWithLineNumbers));
```

Comment:
I think it's unnecessary to copy the map AND make it unmodifiable, but i seriously doubt this is going to be a performance issue ever